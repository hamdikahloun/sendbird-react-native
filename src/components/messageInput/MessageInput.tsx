import React, {memo, useRef} from 'react';
import {StyleSheet, TextInput, View} from 'react-native';
import {MessageTextInput} from './MessageTextInput';
import {SendButton} from './SendButton';

type MessageInputProps = {
  onChangeMessage?: (message: string) => void;
  onSendButtonPressed: (message: string) => void;
  messageTextInputRef?: React.MutableRefObject<TextInput | undefined>;
};
export const MessageInputMemorized = ({
  onChangeMessage,
  onSendButtonPressed,
  messageTextInputRef,
}: MessageInputProps) => {
  const message = useRef('');
  const onMessageChangeInternHandler = (messageText: string) => {
    message.current = messageText;
    onChangeMessage?.(messageText);
  };

  const onSendButtonPressedInternHandler = () => {
    onSendButtonPressed(message.current);
  };

  return (
    <View style={styles.container}>
      <MessageTextInput
        messageTextInputRef={messageTextInputRef}
        onChangeMessage={onMessageChangeInternHandler}
      />
      <SendButton onSendButtonPressed={onSendButtonPressedInternHandler} />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    alignItems: 'center',
    margin: 8,
  },
});
/**
 * MessageInput - UI Component for message input
 *
 * See:
 *
 * [Channel](https://sendbird.com/docs/chat/v3/platform-api/guides/channel-types),
 * [Send your first message](https://sendbird.com/docs/chat/v3/javascript/quickstart/send-first-message)
 * @example
 *
 * const messageTextInputRef = useRef<TextInput>();
 *
 * const onSendButtonPressed = (message: string) => {
 *   sendMessage(message)
 *   messageTextInputRef.current?.clear();
 * };
 *
 * <MessageInput
 *   messageTextInputRef={messageTextInputRef}
 *   onChangeMessage={onChangeMessage}
 *   onSendButtonPressed={onSendButtonPressed}
 * />
 */
export const MessageInput = memo(MessageInputMemorized);
