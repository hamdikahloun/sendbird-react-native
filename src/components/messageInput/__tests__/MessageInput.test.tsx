import React from 'react';
import {render} from '@/../testing';
import {MessageInput} from '../MessageInput';

describe('MessageInput', () => {
  it('should render correctly', async () => {
    const onSendButtonPressed = jest.fn();
    const screen = render(
      <MessageInput onSendButtonPressed={onSendButtonPressed} />,
    ).toJSON();
    expect(screen).toMatchSnapshot();
  });
});
