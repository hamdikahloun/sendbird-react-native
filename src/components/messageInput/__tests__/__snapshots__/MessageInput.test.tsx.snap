// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageInput should render correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "margin": 8,
    }
  }
>
  <TextInput
    onChangeText={[Function]}
    placeholder="Aa"
    style={
      Object {
        "backgroundColor": "#EFEFEF",
        "borderColor": "#F9F9F9",
        "borderRadius": 10,
        "borderWidth": 1,
        "flex": 1,
        "height": 50,
        "marginEnd": 8,
        "padding": 10,
      }
    }
    testID="message"
    value=""
  />
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "height": 48,
        "justifyContent": "center",
        "opacity": 1,
        "padding": 10,
        "width": 48,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/assets/images/send/send.png",
        }
      }
    />
  </View>
</View>
`;
